### PR TITLE
Desktop Preview Feature UI

### PR DESCRIPTION
**Pull request by bjtitus**
Part of #13197

This adds UI and basic functionality to switch between Desktop and Mobile previews:

Mobile Preview

iPad Selection

iPhone Selected Landscape

To test:

- View a blog or site page preview
- Try switching between device previews

A few things still to come:

- Need to use the new device preview icon shown in #13197. I believe a new icon needs to be added to gridicons.
- I think we may want to use the [`contentMode` API](https://developer.apple.com/documentation/webkit/wkwebpagepreferences/contentmode) or other User Agent modification to increase the chance of showing the true Desktop styling.

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ScoutHarris** [comment]
Hey @bjtitus .

 I'm seeing a couple things that aren't consistent with the screenshots:

1) On iPad, the popup has a `Mobile` option, on iPhone it has `Desktop`.

Screen Shot 2020-01-17 at 1 10 18 PM

2) On _Page_ preview, I'm not seeing the switch button on the toolbar:

Screen Shot 2020-01-17 at 1 11 07 PM

**Comment by bjtitus** [comment]
@ScoutHarris

> the popup has a `Mobile` option, on iPhone it has `Desktop`.

Looks like the latest screenshot didn't get uploaded. I just edited that in the PR post.

> On Page preview, I'm not seeing the switch button on the toolbar

This is fixed in the latest commit.

File: WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift [hunk 1]
```diff
@@ -105,6 +164,69 @@ class PreviewWebKitViewController: WebKitViewController {
...
+        return .none
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        // Reset our source rect and view for a transition to a new size
+        guard let navigationController = navigationController,
+            let popoverPresentationController = presentedViewController?.presentationController as? UIPopoverPresentationController,
+            popoverPresentationController.presentedViewController is PreviewDeviceSelectionViewController else { return }
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift]: This return should go on a separate line please.

File: WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift [hunk 2]
```diff
@@ -105,6 +164,69 @@ class PreviewWebKitViewController: WebKitViewController {
...
+extension PreviewWebKitViewController {
+    func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
+        if selectedDevice == .desktop {
+            // Change the viewport scale to match a desktop environment
+            webView.evaluateJavaScript("let originalVp = document.querySelector('meta[name=viewport]').cloneNode(true); originalVp.setAttribute('name', 'original_viewport' ); document.querySelector('head').appendChild(originalVp); parent = document.querySelector('meta[name=viewport]'); parent.setAttribute('content','initial-scale=0');", completionHandler: nil)
+        }
+    }
 }
 
 // MARK: NoResultsViewController Delegate
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift]: I know this wasn't added in this PR, but I just noticed - it doesn't look like this NRV has an action button, so this shouldn't be needed. Which means you can also remove `controller.delegate = self` when creating the `NoResultsViewController`.

Comment (bjtitus) [reply WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift]: That's a good point. We're not using it currently for failed loads like we were in the old preview screen.

File: RELEASE-NOTES.txt [hunk 3]
```diff
@@ -1,6 +1,8 @@
 14.1
 -----
 * Fixes a bug that could cause some web page previews to remain unauthenticated even after logging in.
+* Post Preview: Adds new UI for blog and site page previews.
+* Post Preview: Adds preview for Desktop devices to blog and site page previews.
```
[the remark this answers is not in the document]
Comment (bjtitus) [reply RELEASE-NOTES.txt]: 🤦‍♂ I got excited checking off list items, I guess. 🤷‍♂

File: WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift [hunk 4]
```diff
@@ -1,14 +1,22 @@
...
-
     let post: AbstractPost
 
+    private let canPublish: Bool
+
     private weak var noResultsViewController: NoResultsViewController?
 
+    private var selectedDevice: PreviewDeviceSelectionViewController.PreviewDevice = .default {
+        didSet {
+            webView.reload()
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift]: This still reloads on every selection. Maybe something like this?

```
            if selectedDevice != oldValue {
                webView.reload()
            }
```

**Pull request merged**